contributors/design-proposals/selector-generation-for-all.md

```diff
@@ -0,0 +1,71 @@
+Selector Generation for Deployments, ReplicaSet, DaemonSet, StatefulSet and ReplicationController
+=============
+
+# Goals
+Make selector easy to use and less error-prone for Deployments, ReplicaSet, DaemonSet, StatefulSet and ReplicationController. Make `kubectl apply` work well with selector.
+
+# Problem Description
+The field `spec.selector` of Deployments, ReplicaSet, DaemonSet are defaulted from `spec.template.metadata.labels`, if it is unspecified.
+And there is validation to make sure `spec.selector` always selects the pod template.
+The defaulting of selector may prevent from `kubectl apply` from working when updating the `spec.selector` field. e.g. [kubernetes/kubernetes#26202 (comment)](https://github.com/kubernetes/kubernetes/issues/26202#issuecomment-221421254)
```

Comment:
Can we add an "alternatives considered" section to this doc? I would at least like to see what it would look like to fix this problem by e.g. modifying `kubectl apply` instead of every other component :)